### DISCUSSION TITLE
How to cooperate with NES auto-identification

### DISCUSSION THREAD

**Discussion by fakkuyuu** [sanni/cartreader#764] [Need help]
I have about 30 NES files that I was able to dump successfully with manual settings.
How can I update nes.txt to recognize these automatically? Do I just give the logs to someone else? Or can I rewrite the github nes.txt on my own?

**Comment by sanni** [comment]
The auto config works by calculating two CRC32 checksums over 512 byte of data. The first at offset 0x8000 and the second at 0xE000 as seen by the Cart Reader with no mapper selected. And then comparing these checksums to the second number in the nes.txt database:

nes.txt
````
game.nes
91547086,4A776974,4E45555A020101080000000001000001
````
(So in this case 4A776974)

In the log you see the two calculated checksums(from 0x8000 and 0xE000) the Cart Reader searches for in nes.txt.

````
[+] NES/Famicom

Searching database
for 584E57A6 or 18843011...

CRC not found in database
Using manual selection
````

To add auto config to a game you first do several dump attempts and to see if at least one of the checksum always stays the same and then you just replace the second checksum in nes.txt with that checksum.

Once you have edited the nes.txt and tested it to confirm it works you can do a pull request(you have to fork this github to your github account first i think).

**Reply by fakkuyuu** [reply]
I'll try
